src/main/java/org/gwtbootstrap3/extras/summernote/client/ui/base/mixin/SummernoteValidatorMixin.java

```diff
@@ -0,0 +1,223 @@
+package org.gwtbootstrap3.extras.summernote.client.ui.base.mixin;
+
+/*
+ * #%L
+ * GwtBootstrap3
+ * %%
+ * Copyright (C) 2013 - 2015 GwtBootstrap3
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.TreeSet;
+
+import org.gwtbootstrap3.client.ui.form.error.ErrorHandler;
+import org.gwtbootstrap3.client.ui.form.validator.BlankValidator;
+import org.gwtbootstrap3.client.ui.form.validator.HasValidators;
+import org.gwtbootstrap3.client.ui.form.validator.ValidationChangedEvent;
+import org.gwtbootstrap3.client.ui.form.validator.ValidationChangedEvent.ValidationChangedHandler;
+
+import com.google.gwt.editor.client.EditorError;
+import com.google.gwt.event.logical.shared.ValueChangeEvent;
+import com.google.gwt.event.logical.shared.ValueChangeHandler;
+import com.google.gwt.event.shared.GwtEvent;
+import com.google.web.bindery.event.shared.EventBus;
+import com.google.web.bindery.event.shared.HandlerRegistration;
+import com.google.web.bindery.event.shared.SimpleEventBus;
+import org.gwtbootstrap3.client.ui.form.validator.Validator;
+import org.gwtbootstrap3.client.ui.form.validator.ValidatorWrapper;
+import org.gwtbootstrap3.extras.summernote.client.event.SummernoteOnBlurEvent;
+import org.gwtbootstrap3.extras.summernote.client.event.SummernoteOnBlurHandler;
+import org.gwtbootstrap3.extras.summernote.client.ui.base.SummernoteBase;
+
+/**
+ * Summernote validator mixin. Contains all of the validation logic.
+ * 
+ * @author Ben Dol
+ */
+public class SummernoteValidatorMixin implements HasValidators<String> {
```

Comment:
Oops I was mean't to do that, sorry we had 2 earthquakes like 40 minutes ago so I just started committing my changes haha.